### DISCUSSION TITLE
How do I specify default client settings in ltsp.conf -- for machines not otherwise specified?

### DISCUSSION THREAD

**Discussion by bgpublic** [ltsp/ltsp#661] [Q&A]
I have a situation where I want about a dozen clients to automatically log-in and with a specific username and password, and then I want about 2-dozen machines (and anything new I connect in the future) to log-in with a generic/shared username and password.   In other words, I want a catch-all configuration for all clients that are not otherwise specified in my ltsp.conf.

My current configuration (example below) does not work.   The individually-specified clients will log-in properly, but the lightdm login appears for any client not specified (the auto-login setting is not applied).   If I use a glob on the MAC address " [*] ", that makes all the machines (even individually-specified ones) log-in as defaultuser (although the hostname setting does still work).   I've also tried with the default section at the bottom of the client section, and that did not help.

Is there a way to specify an "else" condition for client configuration in ltsp.conf??

```
# DEFAULT FOR CLIENTS NOT SPECIFIED BELOW:
AUTOLOGIN=defaultuser
RELOGIN=0
PASSWORDS_defaultuser="defaultuser/**********"

[9C:8E:99:D5:40:B3]
HOSTNAME=tc1
AUTOLOGIN=tc1
RELOGIN=0
PASSWORDS_tc1="tc1/**********"

[9C:8E:99:D5:40:A3]
HOSTNAME=tc2
AUTOLOGIN=tc2
RELOGIN=0
PASSWORDS_tc2="tc2/**********"

...
```

**Comment by alkisg** [comment]
I copy paste your example, and I just add a `[clients]` line on top. This then should work:

```
[clients]
# DEFAULT FOR CLIENTS NOT SPECIFIED BELOW:
AUTOLOGIN=defaultuser
RELOGIN=0
PASSWORDS_defaultuser="defaultuser/**********"

[9C:8E:99:D5:40:B3]
HOSTNAME=tc1
AUTOLOGIN=tc1
RELOGIN=0
PASSWORDS_tc1="tc1/**********"

[9C:8E:99:D5:40:A3]
HOSTNAME=tc2
AUTOLOGIN=tc2
RELOGIN=0
PASSWORDS_tc2="tc2/**********"

...
```

**Reply by bgpublic** [reply]
Ok, thank-you for the fast response, @alkisg!

I went to make that change, and then realized that it would have been redundant to the section-marker (i.e. [server], [common], [clients]), which I knew was already in my file.

I then found the reason why those parameters were not being applied wasn't because the [clients] marker was missing, but because another (unexpected by me) client configuration was in the way: [61:6c:6b:69:73:67].   The reason I did not expect it was because it was part of the template ltsp.conf file, rather than something I had done.

I made the mistake if simply skipping-over that example configuration and inserted all of my client configurations below it.   That worked fine for specific client configurations, but was the incorrect location for my defaults because they were unintentionally applied to that one -- example -- MAC address. 

I have since commented-out that sample line/MAC address and tested everything OK.   I hope this helps if anyone else happens to make the same mistake.

**Comment by sbik1978** [comment]
Hello, 
There is another problem with autologin  (and "manual" log in  to). 
Some user is entered in /etc/httpd.conf.
After launching the client, I cannot log in on behalf of the users listed in /etc/passwd and /etc/shadow of the system for which the image was made. 


Would it be the right decision to create these accounts on the server machine?

**Comment by alkisg** [comment]
> Some user is entered in /etc/httpd.conf.

Do you mean that you have an http service and you want to run it under a specific user account?
If so, create a system account (i.e. with uid < 1000) for that service inside the image.

LTSP by default uses the user (uid >= 1000) accounts from the server, and the system (uid < 1000) accounts from the image.

**Reply by sbik1978** [reply]
Thanks for quick answer.
sorry!!
of cause  i mean,  that
 user is entered in /etc/passwd   and  /etc/shadow    in  image  system   
and it is a human user  -  student with very poor  rights.

**Comment by alkisg** [comment]
User accounts should be created on the server, not in the image.

**Reply by sbik1978** [reply]
Thank you!
But this answer makes a new question:
"Is it possible to make some settings in image such as  Desktop icons  and then continue work with theese settings on a terminal client?"
